### PR TITLE
Remove deprecations

### PR DESCRIPTION
**Pull request by ktbarrett**
Remove everything I could find to have been deprecated.

### CODE REVIEW

**Review by marlonjames — APPROVED** [review]
Looks good to me.
Should we put `.. versionchanged:: 2.0` with removal message to replace the deprecated message?

**Comment by ktbarrett** [comment]
I'm not against it. What does @cocotb/maintainers think?

**Comment by cmarqu** [comment]
> Looks good to me. Should we put `.. versionchanged:: 2.0` with removal message to replace the deprecated message?

I'm pro.

File: cocotb/triggers.py [hunk 1]
```diff
@@ -233,27 +232,16 @@ def __init__(
...
-            Using ``None`` as the *units* argument is deprecated, use ``'step'`` instead.
-
         .. versionchanged:: 1.6
             Support rounding modes.
+
+        .. versionchanged:: 2.0
+            Passing ``None`` as the *units* argument was removed, use ``'step'`` instead.
+
+        .. versionchanged:: 2.0
+            ``time_ps`` paramater was removed, use ``time`` parameter instead.
```
Comment (cmarqu) [inline cocotb/triggers.py:242]:
```suggestion
            The ``time_ps`` parameter was removed, use the ``time`` parameter instead.
```

**Pull request merged**